src/ripple/overlay/PeerReservationTable.h

```diff
@@ -0,0 +1,142 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2019 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_OVERLAY_PEER_RESERVATION_TABLE_H_INCLUDED
+#define RIPPLE_OVERLAY_PEER_RESERVATION_TABLE_H_INCLUDED
+
+#include <ripple/beast/hash/uhash.h>
+#include <ripple/core/DatabaseCon.h>
+#include <ripple/json/json_forwards.h>
+#include <ripple/protocol/PublicKey.h>
+
+#define SOCI_USE_BOOST
+#include <boost/optional.hpp>
+#include <soci/soci.h>
+
+#include <string>
+#include <unordered_map>
+
+namespace ripple {
+
+// Value type for reservations.
+// REVIEWER: What is the preferred naming convention for member variables?
```

Comment:
Because I want this to be a value type, ideally one that can be constructed as an aggregate, I'm trying to omit any constructors, which lets me use the nice underscore-less convention without any complications.